### PR TITLE
Finalize flyteidl Rust crate

### PR DESCRIPTION
**Pull request by austin362667**
## Tracking issue

https://github.com/flyteorg/flytekit/pull/2328

## Why are the changes needed?

Expose Flyteidl Rust crate based on https://github.com/flyteorg/flyte/pull/5187

It can be added as a local dependency by Flyekit.

## What changes were proposed in this pull request?

Initialize `cargo.toml` and `lib.rs`.

### Setup process

`cargo b --release`

### Screenshots

Screenshot 2024-04-11 at 6 59 43 PM

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

**Comment by kumare3** [comment]
Will we build a rust client crate here?

**Comment by austin362667** [comment]
I don't think so for now.
We just need to expose `flyteidl` as a Rust Crate in advance. So it can be added as dependency for underlying Rust client.

**Comment by austin362667** [comment]
For example, to add a local dependency, you can do the following:

https://github.com/flyteorg/flytekit/blob/b132c892ee09c1e91d8eeeae8e3d8dd2857a990a/flyrs/Cargo.toml#L20

File: flyteidl/lib.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+pub mod datacatalog {
```
Comment (pingsutw) [inline flyteidl/lib.rs:1]: Are you using datacatalog in flyrs now? if not, should we remove it first?

**Review by pingsutw — APPROVED** [review]

File: flyteidl/lib.rs (hunk 1 above)
Comment (austin362667) [reply flyteidl/lib.rs]: removed!

**Review by pingsutw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Pull request merged**